Fix flaky HttpClientMaxConcurrentStreamTest

received reports for the following flaky test when building with `gradle --parallel clean build` from local
current CI also uses `--parallel` option and hasn't failed yet, but should investigate why this happened. 

```
exceededMaxStreamsPropagatesFailureCorrectly()
java.lang.AssertionError: 
Expected size:<19> but was:<17> in:
<[{connectionAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585486), connectionAcquisitionDuration=11739µs(11739459ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585493), pendingAcquisitionDuration=11727µs(11727314ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585434), connectionAcquisitionDuration=15207µs(15207833ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585440), pendingAcquisitionDuration=15193µs(15193502ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578827), connectionAcquisitionDuration=35863µs(35863344ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578846), pendingAcquisitionDuration=35838µs(35838911ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578926), connectionAcquisitionDuration=38834µs(38834375ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578936), pendingAcquisitionDuration=38307µs(38307104ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585377), connectionAcquisitionDuration=35240µs(35240807ns), socketConnectStartTime=2020-02-18T03:26:01.603Z(1581996361603879), socketConnectDuration=10753µs(10753924ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585384), pendingAcquisitionDuration=18484µs(18484850ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585307), connectionAcquisitionDuration=42649µs(42649641ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585318), pendingAcquisitionDuration=42634µs(42634408ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585134), connectionAcquisitionDuration=43189µs(43189768ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.585Z(1581996361585161), pendingAcquisitionDuration=43160µs(43160134ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578986), connectionAcquisitionDuration=49793µs(49793084ns), socketConnectStartTime=2020-02-18T03:26:01.619Z(1581996361619083), socketConnectDuration=8840µs(8840534ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.578Z(1581996361578993), pendingAcquisitionDuration=40080µs(40080529ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579038), connectionAcquisitionDuration=59512µs(59512222ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579045), pendingAcquisitionDuration=59500µs(59500543ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579092), connectionAcquisitionDuration=62243µs(62243243ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579099), pendingAcquisitionDuration=62226µs(62226715ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582719), connectionAcquisitionDuration=60362µs(60362363ns), socketConnectStartTime=2020-02-18T03:26:01.628Z(1581996361628544), socketConnectDuration=8991µs(8991803ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582726), pendingAcquisitionDuration=45812µs(45812614ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582658), connectionAcquisitionDuration=66369µs(66369322ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582668), pendingAcquisitionDuration=66355µs(66355084ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582552), connectionAcquisitionDuration=66976µs(66976377ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.582Z(1581996361582570), pendingAcquisitionDuration=66954µs(66954147ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579145), connectionAcquisitionDuration=72877µs(72877287ns), socketConnectStartTime=2020-02-18T03:26:01.642Z(1581996361642837), socketConnectDuration=6151µs(6151724ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579152), pendingAcquisitionDuration=63675µs(63675433ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579195), connectionAcquisitionDuration=78081µs(78081725ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579202), pendingAcquisitionDuration=78070µs(78070274ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579243), connectionAcquisitionDuration=78783µs(78783028ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.579Z(1581996361579250), pendingAcquisitionDuration=78771µs(78771213ns)},
    {connectionAcquisitionStartTime=2020-02-18T03:26:01.580Z(1581996361580323), connectionAcquisitionDuration=78745µs(78745771ns), socketConnectStartTime=2020-02-18T03:26:01.651Z(1581996361651400), socketConnectDuration=5531µs(5531173ns), pendingAcquisitionStartTime=2020-02-18T03:26:01.581Z(1581996361581915), pendingAcquisitionDuration=69473µs(69473803ns)}]>
	at com.linecorp.armeria.client.HttpClientMaxConcurrentStreamTest.exceededMaxStreamsPropagatesFailureCorrectly(HttpClientMaxConcurrentStreamTest.java:261)
```

--- a/core/src/test/java/com/linecorp/armeria/client/HttpClientMaxConcurrentStreamTest.java
+++ b/core/src/test/java/com/linecorp/armeria/client/HttpClientMaxConcurrentStreamTest.java
@@ -255,6 +255,8 @@ public class HttpClientMaxConcurrentStreamTest {
         });
 
         await().untilAsserted(() -> assertThat(responses).hasSize(numRequests));
+        await().until(() -> receivedResponses.stream().filter(CompletableFuture::isCompletedExceptionally)
+                                             .count() == numFailedRequests);
         assertThat(opens).hasValue(numExpectedConnections);
         assertThat(connectionTimings.stream().filter(
                 timings -> timings.pendingAcquisitionDurationNanos() > 0))

